tests/unit/test_x509.cpp

```diff
@@ -603,7 +604,7 @@ TEST_F(X509Test, testClientCertificateIsNotCA)
 
 TEST_F(X509Test, testExpiredCertificateIsNotCA)
 {
```

Comment:
I find this test very confusing as it suggests that an expired CA certificate is not marked as CA anymore. However, this is not tested here _root1_expired is an experired (client) certificate issued by root1-CA - so it never was a CA in the first place. I would assume that also invalid certificates (because they are not valid yet or already invalid) are (correctly) marked as CA. Please adapt the test. I think among these "yearXXXX" certificates, there should be suitable ones (maybe none which is expired but some that are not valid yet).

Reply:
Adapted test as suggested. It was necessary to add a new certificate that was both a CA and expired because none of the yearXXXX are expired (year in file name refer to the notBefore time).